src/client/dfs/dfs.c

```diff
@@ -734,6 +734,15 @@ open_file(dfs_t *dfs, daos_handle_t th, dfs_obj_t *parent, int flags,
 		return daos_der2errno(rc);
 	}
 
+	if (flags & O_TRUNC) {
+		rc = daos_array_set_size(file->oh, th, 0, NULL);
+		if (rc) {
+			D_ERROR("Failed to truncate file (%d)\n", rc);
+			daos_array_close(file->oh, NULL);
```

Comment:
I'm slightly confused why this is calling daos_array_close() where on L696 it calls daos_obj_close()

Reply:
you're right. that's actually a bug. those need to be array_close. i'll fix them here and check for possible other occurences.

Reply:
I wonder if we could tweak these types to help flush out such issues?

Reply:
heh.. 
long time ago (3-4 years) there was a suggestion from stakeholders that our daos_handle_t api can programmatically cause bugs like this one since they are not caught at compile time. we said that we will explore or maybe do change to have some explicit types for each type of handles (obj, cont, pool, etc.) instead of daos_handle_t for everything.. but no one acted on that and it's safe to say maybe too late now..

Reply:
We even use it internally in VOS though I do typically have a magic number of something in the data structure the handle references for verification.